homeassistant/components/roku.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for roku platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/roku/
+"""
+import logging
+
+import voluptuous as vol
+
+from typing import Sequence, TypeVar, Union
+from homeassistant.components.discovery import SERVICE_ROKU
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-roku==3.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'roku'
+
+SERVICE_SCAN = 'roku_scan'
+
+ATTR_ROKU = 'roku'
+
+DATA_ROKU = 'data_roku'
+DATA_ENTITIES = 'data_roku_entities'
+
+DEFAULT_NAME = 'Roku'
+
+NOTIFICATION_ID = 'roku_notification'
+NOTIFICATION_TITLE = 'Roku Setup'
+NOTIFICATION_SCAN_ID = 'roku_scan_notification'
+NOTIFICATION_SCAN_TITLE = 'Roku Scan'
+
+T = TypeVar('T')
+
+
+# This version of ensure_list interprets an empty dict as no value
+def ensure_list(value: Union[T, Sequence[T]]) -> Sequence[T]:
+    """Wrap value in list if it is not one."""
+    if value is None or (isinstance(value, dict) and not value):
+        return []
+    return value if isinstance(value, list) else [value]
```

Comment:
Don't copy past. make a function the return a empty list if None and use `vol.All(cv.ensure_list, my_funct)`

Reply:
@pvizeli Do you have an example of a component that does this?

Reply:
`def bla(x): if x is None -> Return [] else x`